Marlin/stepper.cpp

```diff
@@ -492,22 +472,31 @@ ISR(TIMER1_COMPA_vect) {
     // Check endstops
     if (check_endstops) {
       
-      #define _ENDSTOP(axis, minmax) axis ##_## minmax ##_endstop
+      char current_endstop_bits;
+      #ifdef Z_DUAL_ENDSTOPS
+        char current_dual_endstop_bits;
```

Comment:
We have:
 X_MIN
 X_MAX
 Y_MIN
 Y_MAX
 Z_MIN
 Z_MAX
 Z_PROBE
 Z2_MIN
 Z2_MAX

so i count 9 :)